packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -602,29 +671,63 @@ export class StatsAnalyzer extends EventsScope {
             `StatsAnalyzer:index#compareLastStatsResult --> No video RTP packets sent`,
             currentStats.totalPacketsSent
           );
-        } else {
-          if (
-            currentStats.framesEncoded === previousStats.framesEncoded ||
-            currentStats.framesEncoded === 0
-          ) {
-            LoggerProxy.logger.info(
-              `StatsAnalyzer:index#compareLastStatsResult --> No video Frames Encoded`,
-              currentStats.framesEncoded
+        }
+        if (
+          currentStats.framesEncoded === previousStats.framesEncoded ||
+          currentStats.framesEncoded === 0
+        ) {
+          LoggerProxy.logger.info(
+            `StatsAnalyzer:index#compareLastStatsResult --> No video Frames Encoded`,
+            currentStats.framesEncoded
+          );
+          if (this.lastEmittedStartStopEvent.video.local === EVENTS.LOCAL_MEDIA_STARTED) {
+            this.emit(
+              {
+                file: 'statsAnalyzer',
+                function: 'compareLastStatsResult',
+              },
+              EVENTS.NO_VIDEO_ENCODED,
+              {
+                mediaType: 'video',
+              }
             );
           }
+        }
 
-          if (
-            this.statsResults.resolutions['video-send'].send.framesSent ===
-              this.lastStatsResults.resolutions['video-send'].send.framesSent ||
-            this.statsResults.resolutions['video-send'].send.framesSent === 0
-          ) {
-            LoggerProxy.logger.info(
-              `StatsAnalyzer:index#compareLastStatsResult --> No video Frames sent`,
-              this.statsResults.resolutions['video-send'].send.framesSent
+        if (
+          this.statsResults.resolutions['video-send'].send.framesSent ===
+            this.lastStatsResults.resolutions['video-send'].send.framesSent ||
+          this.statsResults.resolutions['video-send'].send.framesSent === 0
+        ) {
+          LoggerProxy.logger.info(
+            `StatsAnalyzer:index#compareLastStatsResult --> No video Frames sent`,
+            this.statsResults.resolutions['video-send'].send.framesSent
+          );
+        }
+
+        // Video is encoded but frames are not sent
+        if (
+          currentStats.framesEncoded !== previousStats.framesEncoded &&
```

Comment:
suggestion: we could avoid duplicating this big condition in this "if" with code from above by just rearranging the conditions, something like this:
```
if (no frames sent) {
   if (no frames encoded) {
    // no frames encoded (so obviously also no frames sent)
   } else {
    // frames encoded but not sent
  }
} 
```

Reply:
Revamped the code.